### ISSUE TITLE
Recipe #154: Locate a Manifest on a Web Map

### ISSUE THREAD

**Issue by glenrobson** [IIIF/trc#92]
## Links

 * Original Issue: https://github.com/IIIF/cookbook-recipes/issues/154
 * Pull Request: https://github.com/IIIF/cookbook-recipes/pull/160
 * Preview: https://preview.iiif.io/cookbook/0154-geo-extension/recipe/0154-geo-extension/

## Background and Summary

This recipe introduces the `navPlace` and shows how it can be used on a Manifest. The example is a statue and the location is where it is currently held but as noted in the recipe the semantic meaning of the location can't be modelled using navPlace as navPlace is only designed for the presentation of the resource. 

## Voting and changes

We welcome comments on the recipe and as well as voting +1, confused face or -1 feel free to add comments to this issue. If this issue is approved then the author will take account of the comments before we merge the branch in to the master cookbook branch.

If the recipe is rejected by the TRC then we will make the changes requested and resubmit it to a future TRC meeting. If you feel that your comments are substantial enough that the recipe should be looked at again by the TRC after the changes have been made please vote -1 (thumbs down). A confused face is treated as abstaining. 

Changes to the recipe will only be made after the TRC voting process has concluded.

**Comment by triplingual** [comment]
On this current reading of the recipe, I'm feeling like `navPlace` needs to be linked out sooner. It's a little tricky to link CODE elements clearly with our current CSS so they are clearly linked, though, so I don't know how to suggest you do it. I think it's also good to have the longer mention linked, but as I skimmed it for a refresher today, I found myself wanting to get out to a broader `navPlace` quicker.

**Comment by triplingual** [comment]
I'm also feeling a preference for reversing the clauses in the LatLong mention:
From
> Note that [Google Maps will display coordinates in Latitude, Longitude order](https://developers.google.com/maps/documentation/javascript/reference/coordinates), but [GeoJSON specifies they need to be recorded in Longitude, Latitude order](https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.1).

to

> Note that [GeoJSON requires coordinates to be recorded in Longitude, Latitude order](https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.1), but [Google Maps will display coordinates in Latitude, Longitude order](https://developers.google.com/maps/documentation/javascript/reference/coordinates).

This foregrounds GeoJSON as the important reference point and segues into the following sentence well.

**Comment by triplingual** [comment]
Also (I promise I'll shut up after this), the links out to the IIIF Extensions and the navPlace guide each imply completeness:
> For all the information on how to use the navPlace property

> For a complete guide on how to use the navPlace property

I'm sure the only difference is not the example in the guide, but textually it ends up looking that way.

**Comment by robcast** [comment]
I like it 👍  

With a suggestion to add something about viewer support i.e. that standard viewers like Mirador and UV do not show anything of the navPlace and have a link to the leaflet viewer once it is public.

**Comment by thehabes** [comment]
> On this current reading of the recipe, I'm feeling like `navPlace` needs to be linked out sooner...

I think that's a fair individualized thought, but only people slightly more comfortable with technical writing will feel that way.  Visual learners probably want an embedded visual or link to an example quicker.  Developers just trying to get the Manifest right will probably want the manifest.json as soon as possible to C & P and make a derivative.

As you noticed, there is a very early mention of `navPlace` that could be linked, but we would have to unCODE it.

**Comment by stephenwf** [comment]
I have a question, can an external geoJSON document be linked, instead of having it inline?

**Comment by kirschbombe** [comment]
It feels like the Implementation Notes section is missing crucial information that tells the reader how to implement. Instead, it refers the reader to the extension to determine how to implement. however, as with all other recipes, a reader should be able to implement the stated use case using only the recipe. I think a reader would expect to see which properties, values, etc. are required/expected in the `navPlace` property to make this use case work.

I recommend replacing "For all the information on how to use the navPlace property see the [implementation notes in the IIIF Extensions directory.](https://iiif.io/api/extension/navplace/#5-implementation-notes)" with a fuller implementation discussion specific to this use case that explains the bits that show up in this manifest.

**Comment by thehabes** [comment]
> I have a question, can an external geoJSON document be linked, instead of having it inline?

Yes, it can be referenced as well so long as what you are referencing is a single Feature Collection.  See https://iiif.io/api/extension/navplace/#21-navplace-property the bit about navPlace values.  There is a small JSON snippet there as an example.

**Comment by thehabes** [comment]
> It feels like the Implementation Notes section is missing crucial information that tells the reader how to implement...

Stating it as simple as possible, you implement this solution by minting a Feature Collection that has all the geographic features you would like to have represent the Manifest.  `navPlace` expects a valid Feature Collection, so definitely use that property.  Web maps expect a valid Feature Collection, so they will draw the features in `manifest.navPlace`.  No matter what resource type, this is what you are doing every time.  The value can only ever be a single Feature Collection.  The recipe avoids noting Feature Collection specifics like the extension spec https://iiif.io/api/extension/navplace/#222-feature-collection.  I think you are hinting at you want that "Feature Collection" section in here.  

The reason it is done on the Manifest is because you are trying to "Locate a Manifest on a Web Map" and not "Locate a Canvas on a Web Map".  The solution for each recipe is the same Feature Collection but a different resource type with the `navPlace` property.

**Comment by thehabes** [comment]
And if your wondering "well what if the Manifest contains Ranges and Canvases all with navPlace"?

This is the viewers job to determine.  Right now if I gave our example viewer an object like this where the Manifest and Canvases in the Manifest have `navPlace` properties, we have it draw all Feature Collections.  [So if the Manifest has a Point and the Canvas has a Point, then you would see two points](https://centerfordigitalhumanities.github.io/navplace-viewer/?iiif-content=https://centerfordigitalhumanities.github.io/navplace-viewer/tests/fixtures/manifest_and_canvases.json).  The problem is there is no configuration to say "I actually don't want that", I am not sure what the default choice should be for viewers in this scenario.  However, that is a Web Map problem not a IIIF problem.

**Comment by kirschbombe** [comment]
@thehabes - yes, mention of the Feature Collection info would be good. We discussed in the Cookbook meeting today and think it would be better if recipes stood on their own a bit more than this one currently does. Adding some brief text about navPlace using GeoJson, requires id and type "Feature Collection", which contains a list of Features, which is where you put your geometry… and maybe some mention of feature properties. It might be good if this appeared higher in the Implementation note section.

I'm voting +1 with the above comments addressed.

**Comment by thehabes** [comment]
Changes are being tracked in https://github.com/IIIF/cookbook-recipes/pull/323 and can be previewed at https://preview.iiif.io/cookbook/0154-post-trc-update/recipe/0154-geo-extension/

**Comment by glenrobson** [comment]
### Issue 92 (Recipe #154: Locate a Manifest on a Web Map)
  +1: 21 [akrishnan15 dlpierce eliotjordan hadro irv jpadfield jtweed julsraemy kirschbombe ksclarke markpatton markpbaggett mikeapp mixterj mposton-folger nfreire omaeazusa regisrob rentonsa thehabes triplingual]
   0: 0 []
  -1: 0 []
  Not TRC: 1 [ViBo448]
  Ineligible: 0 []

### Result: 21 / 21 = 1.00
Super majority is in favor, issue is approved
